servicetalk-grpc-netty/src/test/java/io/servicetalk/grpc/netty/ProtocolCompatibilityTest.java

```diff
@@ -485,6 +493,52 @@ void grpcJavaToServiceTalkBlocking(
         testRequestResponse(client, server, streaming, compression);
     }
 
+    @Test
+    void clientH2ReturnStatus() throws Exception {
+        try (HttpServerContext server = HttpServers.forAddress(localAddress(0))
+                .protocols(h2Default())
+                .listenBlockingAndAwait((ctx, request, responseFactory) -> {
+                    byte meta = request.payloadBody().readByte();
+                    if (meta != 0) {
+                        throw new IllegalArgumentException("compression not supported");
+                    }
+                    int length = request.payloadBody().readInt();
+                    if (request.payloadBody().readableBytes() != length) {
+                        throw new IllegalArgumentException("payload body length incomplete: " + length);
+                    }
+                    CompatRequest compatRequest = CompatRequest.parser().parseFrom(
+                            Buffer.asInputStream(request.payloadBody()));
+                    return responseFactory.newResponse(HttpResponseStatus.of(compatRequest.getId(), "foo"))
+                            .payloadBody(ctx.executionContext().bufferAllocator().fromAscii("error"));
+                });
+             CompatClient grpcJavaClient = grpcJavaClient(server.listenAddress(), null, false, null);
+             CompatClient stClient = serviceTalkClient(server.listenAddress(), false, null, null)) {
+            for (int httpCode = 100; httpCode < 999; ++httpCode) {
+                CompatRequest request = CompatRequest.newBuilder().setId(httpCode).build();
+                int grpcJavaCode = Code.OK.getNumber();
+                int stCode = GrpcStatusCode.OK.value();
+                try {
+                    grpcJavaClient.scalarCall(request).toFuture().get();
+                } catch (ExecutionException e) {
+                    grpcJavaCode = ((StatusRuntimeException) e.getCause()).getStatus().getCode().value();
+                }
+                try {
+                    stClient.scalarCall(request).toFuture().get();
+                } catch (ExecutionException e) {
+                    stCode = ((GrpcStatusException) e.getCause()).status().code().value();
+                }
+                if (httpCode < 200) {
+                    // grpc-java maps 1xx responses to error code INTERNAL, we currently map to UNKNOWN. The test server
+                    // isn't following the http protocol by returning only a 1xx response and each framework catches
+                    // this exception differently internally.
+                    assertThat("mismatch for h2 response code: " + httpCode, stCode, equalTo(UNKNOWN.value()));
```

Comment:
1. Consider adding assertion for grpc status code too:
```asciidoc
assertThat("mismatch for h2 response code: " + httpCode, grpcJavaCode, equalTo(INTERNAL.value()));
```
2. Can you please clarify what prevents us from mapping it to `INTERNAL` too?

Reply:
(1) -> done

(2) -> when there is a 1xx response w/out an expect request our client ignore the initial response headers, splicer sees a NettyBuffer and tries to cast it to HttpResponseMetaData (fails with ClassCastException) which gets mapped to unknown via `DefaultGrpcClientBuilder$CatchAllHttpClientFilter`.